src/content-script/listeners/player.ts

```diff
@@ -47,40 +49,68 @@ function findVideoPlayer(): HTMLVideoElement | null {
 //------------------------------------------------------------------------------
 
 function addVideoListeners(player: HTMLVideoElement) {
-  // TODO: hostRequired should default to true once that's implemented
-  function sendVideoEvent(socketEvent: SocketEvents, hostRequired = false) {
+  // TODO: Once implemented, include communism/facism check
+  function sendVideoEvent(socketEvent: VideoSocketEvents, hostRequired = true) {
     return (event: any) => {
       const state = store.getState();
-      const { isHost } = state.party;
-      const { id: partyId } = state.party;
-      const { currentTime, duration, playbackRate } = event.target;
+      const { isHost, hash } = state.party;
 
-      const payload = { partyId, currentTime, duration, playbackRate };
+      // if party not created, don't send anything
+      if (!hash) return;
 
+      // construct payload
+      const { paused, currentTime, duration, playbackRate } = event.target;
+      const eventData: VideoEventData = { eventType: socketEvent, isHost, paused, currentTime, playbackRate, duration };
+      const payload: VideoEvent = { partyHash: hash, eventData };
+
+      // send payload
       if (hostRequired && isHost) {
         debug(`Host emitting ${socketEvent}`);
-        socket.emit(socketEvent, payload);
+        socket.emit(SocketEvents.VIDEO_EVENT, payload);
       } else if (!hostRequired) {
         debug(`User emitting ${socketEvent}`);
-        socket.emit(socketEvent, payload);
+        socket.emit(SocketEvents.VIDEO_EVENT, payload);
       }
     };
   }
 
-  player.onplay = sendVideoEvent(SocketEvents.VIDEO_PLAY);
-  player.onpause = sendVideoEvent(SocketEvents.VIDEO_PAUSE);
-  player.onseeked = sendVideoEvent(SocketEvents.VIDEO_SEEKED);
-  player.onprogress = sendVideoEvent(SocketEvents.VIDEO_PROGRESS, false);
+  player.onplay = sendVideoEvent(VideoSocketEvents.VIDEO_PLAY);
+  player.onpause = sendVideoEvent(VideoSocketEvents.VIDEO_PAUSE);
+  player.onseeked = sendVideoEvent(VideoSocketEvents.VIDEO_SEEKED);
+  player.onprogress = sendVideoEvent(VideoSocketEvents.VIDEO_PROGRESS, false);
```

Comment:
There's no point to setting `isHost` to `false` if we do a check below and don't handle the video event if it's not from the host, right?